src/ffi/pystate.rs

```diff
@@ -1,26 +1,46 @@
 use crate::ffi::moduleobject::PyModuleDef;
 use crate::ffi::object::PyObject;
-use crate::ffi::PyFrameObject;
 use std::os::raw::{c_int, c_long};
 
 pub const MAX_CO_EXTRA_USERS: c_int = 255;
 
+opaque_struct!(PyThreadState);
 opaque_struct!(PyInterpreterState);
 
-#[repr(C)]
-#[derive(Copy, Clone)]
-pub struct PyThreadState {
-    pub ob_base: PyObject,
-    pub interp: *mut PyInterpreterState,
-}
-
 extern "C" {
+    #[cfg(not(PyPy))]
+    #[cfg_attr(docsrs, doc(not(PyPy)))]
```

Comment:
```suggestion
    #[cfg_attr(docsrs, doc(cfg(not(PyPy))))]
```
Same for all the other ones. 

Maybe this is worth putting on top of the `extern "C"` blocks (splitting them if necessary) to reduce littering attributes everywhere?

Note that doc_cfg doesn't pwork with glob imports, so documenting everything in ffi with these is going to be a huge pain.

Reply:
Hmm. For now I'm going to keep it like this, but I agree it would be nice to find a long-term solution.